libs/sdk-ui/src/base/react/usePromiseCache.ts

```diff
@@ -68,14 +74,25 @@ export function usePromiseCache<TParams, TResult, TError>(
                 setLoading(false);
                 if (usedPromiseCache === promiseCacheRef.current) {
                     setResults(results);
+                    shouldResetPromisesRef.current = true;
                 }
             })
             .catch((errors) => {
+                if (errors?.message === "Canceled") {
+                    return;
+                }
                 setLoading(false);
                 if (usedPromiseCache === promiseCacheRef.current) {
                     setErrors(errors);
                 }
             });
+
+        return () => {
+            // this shouldResetPromises flag is only apply for the InsightList.tst component
+            // to prevent the promises is reset before be canceled.
+            shouldResetPromisesRef.current = fetchDeps.length < MAX_FETCH_LENGTH;
```

Comment:
ah, I just realized this: is it that you recognize this fetchDeps.length from 
```
        [backend, pagesToLoad, search, selectedTabId, insightListLastUpdateRequested],
```
as the list has 5 items? If so, it looks very fragile to me. Could we introduce some reset flag option for the hook so that we switch resetting/non-resetting behaviour and just set this option to true just from the insightlist?

Reply:
I have added a new flag for this hook. Thank you!